engine/common/api_helpers/custom_fields.py

```diff
@@ -102,25 +100,6 @@ def to_internal_value(self, data):
         return queryset.filter(organization=request.user.organization, public_primary_key__in=data).distinct()
 
 
-class CustomTimeField(fields.TimeField):
-    def to_representation(self, value):
-        result = super().to_representation(value)
-        if result[-1] != "Z":
-            result += "Z"
-        return result
-
-    def to_internal_value(self, data):
-        TIME_FORMAT_LEN = len("00:00:00Z")
-        if len(data) == TIME_FORMAT_LEN:
-            try:
-                time.strptime(data, "%H:%M:%SZ")
-            except ValueError:
-                raise BadRequest(detail="Invalid time format, should be '00:00:00Z'")
-        else:
-            raise BadRequest(detail="Invalid time format, should be '00:00:00Z'")
-        return data
```

Comment:
returning a `str` is the main issue here. If you look at `fields.TimeField` it returns a `time.time` object (which solves the issue seen in [the stack trace](https://github.com/grafana/oncall-private/issues/2373)).

Additionally, we don't need to subclass `fields.TimeField` just to validate the time format, we can use the `format` and `input_format` kwargs instead.